### PR TITLE
Support rendering leading Icon to TextField 

### PR DESCRIPTION
**Pull request by ethanshar**
- Support rendering `leadingIcon` prop
- Prevent rendering of `prefix` or `leadingIcon` with `floatingPlaceholder` 
(WOAUILIB-499)

### CODE REVIEW

File: demo/src/screens/componentScreens/TextFieldScreen/BasicTextFieldScreen.js [hunk 1]
```diff
@@ -57,6 +64,8 @@ export default class BasicTextFieldScreen extends Component {
       error
     } = this.state;
 
+    
+
```
Comment (ArnonZ) [inline demo/src/screens/componentScreens/TextFieldScreen/BasicTextFieldScreen.js:68]: Not needed

File: src/components/inputs/TextField.js [hunk 2]
```diff
@@ -216,7 +220,7 @@ export default class TextField extends BaseInput {
     const translate = width / 2 - (width * FLOATING_PLACEHOLDER_SCALE) / 2;
     this.setState({floatingPlaceholderTranslate: translate / FLOATING_PLACEHOLDER_SCALE});
   };
-
+
```
Comment (ArnonZ) [inline src/components/inputs/TextField.js:223]: tab

File: demo/src/screens/componentScreens/TextFieldScreen/BasicTextFieldScreen.js (hunk 1 above)
Comment (ethanshar) [reply demo/src/screens/componentScreens/TextFieldScreen/BasicTextFieldScreen.js:68]: I think it's a github issue.. 
anyway I remove those lines, maybe it'll fix it..

File: demo/src/screens/componentScreens/TextFieldScreen/BasicTextFieldScreen.js [hunk 3]
```diff
@@ -21,13 +21,19 @@ const GUIDING_TEXTS = {
   floatingPlaceholder: 'Floating Placeholder'
 };
 
+const LEADING_ICON = {
+  source: Assets.icons.demo.search,
+  style: {marginRight: Spacings.s1}
+};
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply demo/src/screens/componentScreens/TextFieldScreen/BasicTextFieldScreen.js:27]: I preferred to keep it clean as long as it doesn't look bad without custom styling. 
This spacing is just for the sake of this example. 
I assume we will pass something else in the private config

File: src/components/inputs/TextField.js [hunk 4]
```diff
@@ -345,8 +348,9 @@ export default class TextField extends BaseInput {
   }
 
   shouldFakePlaceholder() {
-    const {floatingPlaceholder, centered} = this.getThemeProps();
-    return Boolean(floatingPlaceholder && !centered && !this.shouldShowTopError());
+    const {floatingPlaceholder, centered, leadingIcon, prefix} = this.getThemeProps();
+
+    return !leadingIcon && !prefix && Boolean(floatingPlaceholder && !centered && !this.shouldShowTopError());
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/components/inputs/TextField.js:353]: I was about to add it, but it caused a weird UI issue where the floating placeholder transition looked bad. 
Anyway, I discussed with Yulia and she agreed there's no reason for both of these features to come together.

File: src/components/inputs/TextField.js (hunk 2 above)
Comment (ethanshar) [reply src/components/inputs/TextField.js]: fixed

File: src/components/inputs/TextField.js [hunk 5]
```diff
@@ -642,6 +649,7 @@ export default class TextField extends BaseInput {
             {paddingTop: this.getTopPaddings()}
           ]}
         >
+          {leadingIcon && <Image {...leadingIcon} onLayout={this.onLeadingIconLayout}/>}
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/components/inputs/TextField.js]: fixed

**Pull request merged**